### ISSUE TITLE
Training code

### ISSUE THREAD

**Issue by tarashakhurana** [intel-isl/MiDaS#43]
Do you plan to release your training code sometime in the future? It would be really helpful to advance the research on monocular depth estimation!

If not, can you explain how the Pareto optimatility is ensured during training? It seems like there will also have to be an undo step in the training pipeline such that whenever the Pareto optimum is reached and the next backpropagation update disturbs this state, this update will have to be reversed.

**Comment by ranftlr** [comment]
There are no plans to release the training code.

I'm not sure if I understand your question right. The algorithm that we are using is based on multiple gradient descent. As such it converges to a Pareto-stationary point, which means that there is no update if you are in a pareto-optimal point. However, in practice this is only approximately true, as we are working with stochastic approximations of the objectives. Did you have a look at the corresponding paper that covers the algorithm (https://arxiv.org/abs/1810.04650)?

**Comment by Sankar-CV** [comment]
Dear author's,

Thank you so much for such a great work and for sharing the trained model. The proposed MiDaS model works on diverse scenes with significant performance. This is one of the great contributions for the research community. I would like understand two points here:

(1) The depth map inferred from pre-trained MiDaS model is in the form of inverse depth, even after doing inversion I am not able to get absolute depth. What is the method to get absolute depth from the inverse depth?

(2) The quality of depth predicted from pre-trined MiDaS model per frame is really good but it is also inconsistent/jittery over a video sequence. It would be grateful if the training code is also accessible so that MiDaS model can be improved further.

**Comment by vasavamsi** [comment]
Dear Authors,

Can you please provide the train script for the model, as it would be really helpful for see the performance of the model on other datasets.

Regards,

**Comment by ranftlr** [comment]
@Sankar-CV See for example #42 for answers surrounding relative depth.

There are still no plans to release the training code.

**Comment by axbycc-mark** [comment]
This is what I gather from reading the paper and following through the citations.

It looks like the training code would be based off of the code by the author of "Multi-Task Learning as Multi-Objective Optimization" posted here https://github.com/isl-org/MultiObjectiveOptimization.

That paper describes how to do approximate pareto optimization using a single forward and backward pass by fusing the different task gradients using convex optimization. The paper mentions two types of gradients, task-specific and shared. The MiDaS paper mentions no task-specific weights so the repo linked above can be further simplified for anyone trying to re-implement a MiDaS training script. 

For MiDas, a "task" is a dataset. To mix four datasets, you would sample one batch from each dataset, and run them through the net. For the backwards pass, you would use the repo linked above.
